soccer/motion/MotionControl.cpp

```diff
@@ -35,244 +30,150 @@ void MotionControl::createConfiguration(Configuration* cfg) {
+                        MotionSetpoint* setpoint) {
+    if (!setpoint) {
+        reset();
         return;
     }
 
-    if (_robot->isJoystickControlled()) {
-        // We are joystick controlled. Any integral we had before entering
-        // joystick control isn't going to be relevant after we exit, so we
-        // should clear it.
-        _positionXController.clearWindup();
-        _positionYController.clearWindup();
-        _angleController.clearWindup();
+    if (!state.visible || !path.path) {
+        reset();
+        stop(setpoint);
         return;
     }
 
-    const MotionConstraints& constraints = _robot->motionConstraints();
-
-    // update PID parameters
-    _positionXController.kp = *_robot->config->translation.p;
-    _positionXController.ki = *_robot->config->translation.i;
-    _positionXController.setWindup(*_robot->config->translation.i_windup);
-    _positionXController.kd = *_robot->config->translation.d;
-    _positionYController.kp = *_robot->config->translation.p;
-    _positionYController.ki = *_robot->config->translation.i;
-    _positionYController.setWindup(*_robot->config->translation.i_windup);
-    _positionYController.kd = *_robot->config->translation.d;
-    _angleController.kp = *_robot->config->rotation.p;
-    _angleController.ki = *_robot->config->rotation.i;
-    _angleController.kd = *_robot->config->rotation.d;
-
-    RJ::Seconds timeIntoPath =
-        (RJ::now() - _robot->path().startTime()) + RJ::Seconds(1.0 / 60);
-
-    // evaluate path - where should we be right now?
-    std::optional<RobotInstant> optTarget =
-        _robot->path().evaluate(timeIntoPath);
-
-    if (!optTarget) {
-        optTarget = _robot->path().end();
-        _context->debug_drawer.drawCircle(optTarget->motion.pos, .15, Qt::red,
-                                          "Planning");
-    } else {
-        Point start = _robot->pos();
-        _context->debug_drawer.drawCircle(optTarget->motion.pos, .15, Qt::green,
-                                          "Planning");
-    }
-
-    // Angle control //////////////////////////////////////////////////
-    ////////////////////////////////////////////////////////////////////
-
-    float targetW = 0;
-    auto& rotationCommand = _robot->rotationCommand();
-    const auto& rotationConstraints = _robot->rotationConstraints();
-
-    std::optional<Geometry2d::Point> targetPt;
-    const auto& motionCommand = _robot->motionCommand();
-
-    std::optional<float> targetAngleFinal;
-    // if (motionCommand->getCommandType() == MotionCommand::Pivot) {
-    //    PivotCommand command =
-    //    *static_cast<PivotCommand*>(motionCommand.get());
-    //    targetPt = command.pivotTarget;
-    //} else {
-    if (optTarget) {
-        if (optTarget->angle) {
-            if (optTarget->angle->angle) {
-                targetAngleFinal = *optTarget->angle->angle;
-            }
-        }
-    }
-    //}
+    // TODO: Reset and return if robot is joystick-controlled.
```

Comment:
I would make sure this is added before actually merging. It would be a real surprise to whoever tests next if you dont lol

Reply:
Yeah, I'll add that in tomorrow. I just couldn't think of a good place for the "is joystick controlled" field to go in Context.